packages/vhd-lib/src/Vhd/VhdDirectory.js

```diff
@@ -100,14 +100,17 @@ export class VhdDirectory extends VhdAbstract {
     )
     // here we can implement compression and / or crypto
 
-    // chunks can be in sub directories :  create directories if necessary
-    const pathParts = partName.split('/')
-    let currentPath = this._path
-
-    // the last one is the file name
-    for (let i = 0; i < pathParts.length - 1; i++) {
-      currentPath += '/' + pathParts[i]
-      await this._handler.mkdir(currentPath)
+    // s3 don't really have directories, don't lose time emulating it
+    if (this._handler.type !== 's3') {
+      // chunks can be in sub directories :  create directories if necessary
+      const pathParts = partName.split('/')
+      let currentPath = this._path
+
+      // the last one is the file name
+      for (let i = 0; i < pathParts.length - 1; i++) {
+        currentPath += '/' + pathParts[i]
+        await this._handler.mkdir(currentPath)
+      }
     }
     return this._handler.writeFile(this._getChunkPath(partName), buffer, this._opts)
```

Comment:
Why not simply using `outputFile` instead?

Reply:
because I didn't see it, it's a better choice. I update the code immediatly